packages/shared/lib/clients/orchestrator.ts

```diff
@@ -79,6 +82,20 @@ export class Orchestrator {
         return `environment:${environmentId}:sync:${syncId}`;
     }
 
+    async searchSchedules(props: { syncId: string; environmentId: number }[]): Promise<Result<Map<string, OrchestratorSchedule>>> {
+        const scheduleNames = props.map(({ syncId, environmentId }) => this.getScheduleName({ environmentId, syncId }));
+        const schedules = await this.client.searchSchedules({ scheduleNames, limit: scheduleNames.length });
+        if (schedules.isErr()) {
+            return Err(`Failed to get schedules: ${stringifyError(schedules.error)}`);
+        }
+        const scheduleMap = schedules.value.reduce((map, schedule) => {
+            const syncId = schedule.name.split(':').pop()!;
```

Comment:
a bit convoluted but I don't have a better solution.

Reply:
your name seems to follow `property:value` pattern, maybe a unserializer that split and output an object would clarify `environment:2:connection:9` => `{environment: 2, connection: 9}`

Reply:
fe9c4ff51b79d9bd69ae17cc23bca41303f42260